src/core/Heading/Heading.tsx

```diff
@@ -44,6 +46,7 @@ const StyledHeading = styled(
         [smallScreenClassName]: smallScreen,
       })}
       variant={variant === 'h1hero' ? 'h1' : variant}
+      as={asProp}
```

Comment:
I decided to make the internal `asProp` -> `as` implementation here at `src/core`-level component as it has nothing to do with `src/components` level implementations. There it will work just fine to use `as` (supported by styled-components).